models/utility.py

```diff
@@ -0,0 +1,146 @@
+import numpy as np
+from scipy.constants import c as C_c
+from scipy.constants import epsilon_0
+import scipy.constants as const
+
+
+def wavelength2freq(wavelength):
+    return 299792458 / wavelength
+
+
+def freq2wavelength(frequency):
+    return 299792458 / frequency
+
+
+def power2field(laserPower, laserWaist):
+    # P is power in W
+    # A is area in cm^2
+    # E is field in V/m
+    maxIntensity = 2 * laserPower / (np.pi * laserWaist**2)
+    electricField = np.sqrt(2.0 * maxIntensity / (C_c * epsilon_0))
+    return electricField
+
+def p2i_gaussian(P,w0):
+    """
+    convert power to intensity in a gaussian beam
+    
+    Parameters:
+        P: Power of laser (W)
+        w0: waist of laser (m)
+    """
+    return 2 * P / (np.pi * w0**2) # 2P/(pi*w0^2)
+
+def w2omega(wavelength):
+    """
+        Convert wavelength to angular frequency.
+
+        This function calculates the angular frequency of light given its wavelength
+        using the formula: ω = 2πc/λ, where c is the speed of light in vacuum.
+
+        Args:
+            wavelength (float): Wavelength of the light in meters (m).
+
+        Returns:
+            float: Angular frequency in radians per second (rad/s).
+
+    """
+    return 2*np.pi*const.c/wavelength # 2*pi*c/w 
+
+def E2T(U0, debug = False):
```

Comment:
Name the function so we know what it does without needing to consult the docstring.